ibis-server/tests/routers/v2/connector/test_oracle.py

```diff
@@ -0,0 +1,134 @@
+import base64
+
+import orjson
+import pandas as pd
+import pytest
+import sqlalchemy
+from sqlalchemy import text
+from testcontainers.oracle import OracleDbContainer
+
+from tests.conftest import file_path
+
+pytestmark = pytest.mark.oracle
+
+base_url = "/v2/connector/oracle"
+
+manifest = {
+    "catalog": "my_catalog",
+    "schema": "my_schema",
+    "models": [
+        {
+            "name": "Orders",
+            "tableReference": {
+                "schema": "SYSTEM",
+                "table": "ORDERS",
+            },
+            "columns": [
+                {"name": "orderkey", "expression": "O_ORDERKEY", "type": "number"},
+                {"name": "custkey", "expression": "O_CUSTKEY", "type": "number"},
+                {
+                    "name": "orderstatus",
+                    "expression": "O_ORDERSTATUS",
+                    "type": "varchar2",
+                },
+                {
+                    "name": "totalprice",
+                    "expression": "O_TOTALPRICE",
+                    "type": "number",
+                },
+                {"name": "orderdate", "expression": "O_ORDERDATE", "type": "date"},
+                {
+                    "name": "order_cust_key",
+                    "expression": "O_ORDERKEY || '_' || O_CUSTKEY",
+                    "type": "varchar2",
+                },
+                {
+                    "name": "timestamp",
+                    "expression": "CAST('2024-01-01 23:59:59' AS TIMESTAMP)",
+                    "type": "timestamp",
+                },
+                {
+                    "name": "test_null_time",
+                    "expression": "CAST(NULL AS TIMESTAMP)",
+                    "type": "timestamp",
+                },
+                {
+                    "name": "blob_column",
+                    "expression": "UTL_RAW.CAST_TO_RAW('abc')",
+                    "type": "blob",
+                },
+            ],
+            "primaryKey": "orderkey",
+        }
+    ],
+}
+
+
+@pytest.fixture(scope="module")
+def manifest_str():
+    return base64.b64encode(orjson.dumps(manifest)).decode("utf-8")
+
+
+@pytest.fixture(scope="module")
+def oracle(request) -> OracleDbContainer:
+    oracle = OracleDbContainer(
+        "gvenzl/oracle-free:23.6-slim-faststart", oracle_password="Oracle123"
+    )
+
+    oracle.start()
+
+    host = oracle.get_container_host_ip()
+    port = oracle.get_exposed_port(1521)
```

Comment:
```suggestion
    port = oracle.get_exposed_port(oracle.port)
```
I prefer to use the original port to avoid the potential port conflict.